eng/Versions.props

```
@@ -31,7 +31,7 @@
   <PropertyGroup>
     <!-- DotNetCliVersion MUST match the dotnet version in global.json.
          Otherwise, this version of dotnet will not be installed and the build will error out. -->
-    <DotNetCliVersion>6.0.100-preview.3.21202.5</DotNetCliVersion>
+    <DotNetCliVersion>$([System.Text.RegularExpressions.Regex]::Match($([System.IO.File]::ReadAllText('$(MSBuildThisFileDirectory)..\global.json')), '"dotnet": "([^"]*)"').Groups.get_Item(1))</DotNetCliVersion>
```

See if we have `JPath`/`JsonPath` similar to XML's `XPath` in `System.Text.Json`, we could have a nice looking code...

```suggestion
    <DotNetCliVersion>$([System.Text.Json.JsonPath]::GetValueFromFile('global.json', '$.tools.dotnet(.version)?'))</DotNetCliVersion>
```

OR

```suggestion
    <DotNetCliVersion>$([System.Text.Json.JsonPath]::GetValueFromFile('global.json', '$.sdk(.version)?'))</DotNetCliVersion>
```